impala/dbapi.py

```diff
@@ -24,9 +24,7 @@
     Error, Warning, InterfaceError, DatabaseError, InternalError,
     OperationalError, ProgrammingError, IntegrityError, DataError,
     NotSupportedError)
-from impala.util import (
-    warn_deprecate_hs2, warn_deprecate, warn_deprecate_protocol)
-import impala.beeswax as beeswax
+from impala.util import warn_deprecate, warn_deprecate_protocol
```

Comment:
Could you modify `warn_deprecate_protocol` to say that it's no longer necessary because HiveServer2 is the only protocol supported?

And maybe something like:

```python
if protocol.lower() == 'hiveserver2':
    warn_deprecate_protocol()
elif protocol is not None:
    raise NotSupportedError('...')  # or similar
service = hs2.connect(...)
```

And finally, move the `protocol=None` parameter further to the back (with the other deprecated parameters).